packages/sync/test/SquashOperation.test.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  Operation
+} from "apollo-link";
+import { squashOperations } from "../src/offline/squashOperations";
+
+import { OperationQueueEntry } from "../src/links/OfflineQueueLink";
+import { expect } from "chai";
+import { assertAbstractType } from "graphql";
+
+let operation: Operation = {
+  variables: {
+    name: "User 1",
+    id: 5,
+    dateOfBirth: "1/1/18",
+    address: "GraphQL Lane",
+    version: 1
+  },
+  operationName: "updateUser",
+  query: {
+    kind: "Document",
+    definitions: [{
+      kind: "OperationDefinition",
+      selectionSet: {} as any,
+      operation: "mutation",
+      name: {
+        kind: "Name",
+        value: "updateUser"
+      }
+    }]
+  },
+  extensions: {} as any,
+  setContext: {} as any,
+  getContext: {} as any,
+  toKey: {} as any
+};
+
+let operationDifferentQuery: Operation = {
+  variables: {
+    name: "User 1",
+    id: 5,
+    dateOfBirth: "1/1/18",
+    address: "GraphQL Lane",
+    version: 1
+  },
+  operationName: "createUser",
+  query: {
+    kind: "Document",
+    definitions: [{
+      kind: "OperationDefinition",
+      selectionSet: {} as any,
+      operation: "mutation",
+      name: {
+        kind: "Name",
+        value: "createUser"
+      }
+    }]
+  },
+  extensions: {} as any,
+  setContext: {} as any,
+  getContext: {} as any,
+  toKey: {} as any
+};
+
+let queueEntry: OperationQueueEntry = {
+  operation,
+  forward: {} as any,
+  observer: {} as any
+};
+
+let queueEntryDifferentID: OperationQueueEntry = {
+  operation: {...operation, variables: {
+    name: "User 1",
+    id: 44,
+    dateOfBirth: "1/1/18",
+    address: "GraphQL Lane",
+    version: 1
+  }},
+  forward: {} as any,
+  observer: {} as any
+}
+
+let queueEntryDifferentQuery: OperationQueueEntry = {
+  operation: operationDifferentQuery,
+  forward: {} as any,
+  observer: {} as any
+}
+
+describe("SquashOperations", () => {
+  let opQueue: OperationQueueEntry[] = [];
+  beforeEach(() => {
+    opQueue = [];
+  });
+
+  it("check queue contains single operation", () => {
+    const queue = squashOperations(queueEntry, opQueue);
+    expect(queue.length).eqls(1);
```

Comment:
[Very optional ] I guess we can also check if things are merged using object assign but that is an stretch goal.